src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/Contracts/StackWalk/Context/ContextHolder.cs

```
@@ -42,7 +45,63 @@ public unsafe byte[] GetBytes()
         Span<byte> byteSpan = MemoryMarshal.AsBytes(structSpan);
         return byteSpan.ToArray();
     }
-    public IPlatformAgnosticContext Clone() => new CotnextHolder<T>() { Context = Context };
+    public IPlatformAgnosticContext Clone() => new ContextHolder<T>() { Context = Context };
     public void Clear() => Context = default;
     public void Unwind(Target target) => Context.Unwind(target);
+
+    public bool TrySetRegister(Target target, string fieldName, TargetNUInt value)
+    {
+        if (typeof(T).GetField(fieldName) is not FieldInfo field) return false;
+        switch (field.FieldType)
+        {
+            case Type t when t == typeof(ulong) && target.PointerSize == sizeof(ulong):
+                field.SetValueDirect(__makeref(Context), value.Value);
+                return true;
+            case Type t when t == typeof(uint) && target.PointerSize == sizeof(uint):
+                field.SetValueDirect(__makeref(Context), (uint)value.Value);
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    public bool TryReadRegister(Target target, string fieldName, out TargetNUInt value)
+    {
+        value = default;
+        if (typeof(T).GetField(fieldName) is not FieldInfo field) return false;
+        if (field.GetValue(Context) is not object fieldValue) return false;
+        if (fieldValue is ulong ul && target.PointerSize == sizeof(ulong))
```

(not strictly necessary to fix, just a thought)
do we want to communicate something if they are mismatched and fail in a loud way, i.e. 'expected uint but got ulong' or is it enough to just return false? This is 'correct' but I don't know if this is a good opportunity for us to "fail well" by properly detecting and messaging error scenarios at this layer instead of just `return`ing a `false` that conveys no information about the nature of the failure. I'm assuming an end user is touching this code through 20 layers of abstraction, and if this method returns false because the field held a `ulong` instead of a `uint` they might really appreciate it if there were an appropriate exception or error handler for that scenario.
I would assume the only way we can reach the `return false` at the bottom is if the target and the context are somehow mismatched, like the wrong dll was loaded. That's a scenario where we would want to fail fast, I think, instead of just silently not working?
On the other hand, this is robust, so I have no problem with it being checked in.

---

Agreed that we usually want to fail fast. Currently that happens in the caller, it could be moved in here but I'm not sure on other scenarios if that's always the behavior we want.